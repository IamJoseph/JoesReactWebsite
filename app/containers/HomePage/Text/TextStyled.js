import styled from 'styled-components';

export const Container = styled.section`
  align-items: flex-end;
  display: flex;
  flex-direction: column;
  font-size: 1.35em;
  justify-content: flex-start;
  margin-top: 70px;
  width: 100%;
  @media(orientation: landscape){
    align-items: center;
    flex-direction: row;
    justify-content: flex-end;
  }
  @media(min-width: 768px){
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
  }
`;

export const H1 = styled.h1`
  color: #e9de63;
  font-size: 2.2em;
  margin: 0;
`;

/*
 * Text Styles
 */

export const Text = styled.div`
 background-color: rgba(255, 254, 246, .95);
 border-radius: 5px;
 font-size: .8em;
 lineHeight: 1.5;
 margin-left: 2em;
 margin-right: 2em;
 margin-top: ${(props) => (props.right ? '1em' : '4em')};
 padding: 1em;
 user-select: none;
 width: 18em;
 word-wrap: break-word;
 z-index: 2;
 @media(orientation: landscape){
   margin-top: ${(props) => (props.right ? 0 : '-3em')};
 }
 @media(min-width: 375px){
   font-size: .9em;
 }
 @media(min-width: 414px) and (orientation: portrait), (min-width: 700px){
   font-size: 1em;
 }
 @media(min-width: 768px){
   border: 1px solid #686664;
   width: 9em;
   @media(orientation: portrait){
     margin-top: 0;
   }
 }
 @media(min-width: 1150px){
   width: 13em;
 }
 @media(min-width: 1400px){
   background-color: transparent;
   border: none;
   margin-left: 0;
   margin-right: 0;
   margin-top: ${(props) => (props.right ? 0 : '-5em')};
   padding: 1.87em;
   width: ${(props) => (props.right ? '20em' : '23em')};
 }
`;
