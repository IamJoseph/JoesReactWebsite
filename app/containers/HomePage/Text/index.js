import React from 'react';
import { Container, Text, H1 } from './TextStyled';
import Brush from '../../../assets/brush_stroke.png';

export default function TextContainer() {
  return (
    <Container>
      <Text className="textTransform">
        <div style={{ position: 'relative' }}>
          <H1>Hello</H1>
          <img style={{ width: '7.2em', marginTop: '-1.3em' }} alt="brush stroke under hello for emphasis and style" src={Brush} />
        </div>
        I&apos;m Joseph Ireland, a Web Engineer / Software Developer / Jack of all Trades Programmer from Orlando, FL.
      </Text>
      <Text right className="textTransform">
        I &#9825; space, technology, creativity, aesthetically pleasing UI and scalable performant code.
      </Text>
    </Container>
  );
}
