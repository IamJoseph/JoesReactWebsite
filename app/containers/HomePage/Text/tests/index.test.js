import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Text from '../index';

describe('<Text />', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <Text />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
