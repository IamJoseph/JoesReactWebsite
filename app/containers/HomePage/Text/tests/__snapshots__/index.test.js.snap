// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render correctly 1`] = `
<TextContainer>
  <styled.section>
    <section
      className="dzYAfK"
    >
      <styled.div
        className="textTransform"
      >
        <div
          className="textTransform fnBkzA"
        >
          <div
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <styled.h1>
              <h1
                className="fojIok"
              >
                Hello
              </h1>
            </styled.h1>
            <img
              alt="brush stroke under hello for emphasis and style"
              src="IMAGE_MOCK"
              style={
                Object {
                  "marginTop": "-1.3em",
                  "width": "7.2em",
                }
              }
            />
          </div>
          I'm Joseph Ireland, a Web Engineer / Software Developer / Jack of all Trades Programmer from Orlando, FL.
        </div>
      </styled.div>
      <styled.div
        className="textTransform"
        right={true}
      >
        <div
          className="textTransform bMBOrX"
        >
          I â™¡ space, technology, creativity, aesthetically pleasing UI and scalable performant code.
        </div>
      </styled.div>
    </section>
  </styled.section>
</TextContainer>
`;
